docs/man/rpm-scriptlets.7.scd

```diff
@@ -0,0 +1,421 @@
+
+*RPM_INSTALL_PREFIX*
+	Same as *RPM_INSTALL_PREFIX0*, for legacy compatibility.
+
+# EXIT STATUS
+A non-zero exit code from any scriptlet will be reflected as an error in the
+transaction result code. Additionally, non-zero exit from the various
+pre-scriptlets (noted above for each such scriptlet) stop further processing
+related to _that package_, not the whole transaction.
+
+Scriptlets should be designed to always return with zero exit code, as RPM
+cannot undo or roll back a transaction.
+
+For example, an exit code of 1 from a *%pre* script prevents the install/update
+of that package, but the rest of the transaction will still take place,
+even if there are missing dependencies.
+
+# EXAMPLES
+*Example 1. Update /etc/shell on package install/removal*
+
+```
+%post
+if [ $1 -eq 1 ]; then
+    # Initial install
+    grep -q '^/bin/mysh$' /etc/shells || echo '/bin/mysh' >> /etc/shells
+fi
+
+%postun
+if [ $1 -eq 0 ] ; then
+    # Package removal, not upgrade
+    sed -i '/^\/bin\/mysh$/d' /etc/shells
+fi
+```
+
+*Example 2. Manage alternatives in a subpackage*
+
+```
+Name: frobber
+[...]
+
+%package myimpl
+[...]
+
+%post myimpl
+update-alternatives --install /usr/bin/frobber \\
+	frobber /usr/bin/frobber.myimpl 10
+
+%preun myimpl
+if [ $1 -eq 0]; then
+    # Package removal, not upgrade
+    update-alternatives --remove frobber /usr/bin/frobber.myimpl 10
+fi
+```
+
+*Example 3. Manage the MIME database*
+
+```
+%transfiletriggerin -- /usr/share/mime
+/usr/bin/update-mime-database -n /usr/share/mime &> /dev/null ||:
+
+%transfiletriggerpostun -- /usr/share/mime
+/usr/bin/update-mime-database -n /usr/share/mime &> /dev/null ||:
+```
+
+Note that these would go to the package owning the
+_/usr/bin/update-mime-database_ binary, which then centrally handles
+all udpates to the _/usr/share/mime_ directory. Besides removing the need
+to have such scriptlets in all packages with such content, it also eliminates
+the need to have dependencies to the helper binary in all related packages.
+
+*Example 4. Enable a systemd preset on upgrade from older version*
+
+```
+Name: my
+Version: 3.0
+[...]
+
+# On update from 2.x
+%triggerun -- my < 3.0-1
+if [ -x /usr/bin/systemctl ]; then
+    systemctl --no-reload preset mydb-migrate ||:
+fi
+```
+
+*Example 5. Execute a program in %pretrans*
+
+```
+%pretrans -p <lua>
+if posix.access('/usr/bin/mything', 'x') then
+    rpm.execute('/usr/bin/mything')
+end
+```
+
+Note that this seemingly simple thing is generally not possible in
+*%pretrans* without using *-p <lua>*, as on the initial install of
+a system, even _/bin/sh_ is not present at all.
+
+*Example 6. Runtime macro expansion*
+
+```
+%post -e
+ln -s %{_datadir}/bar/some.ext %%{_libdir}/foo/some.ext
+```
+
+*Example 7. Runtime queryformat expansion for installed filenames*
+
+```
+%post -q
+for f in [%%{instfilenames} ]; do
+    echo $f
+done
+```
+
+This will accurately reflect any files skipped due to options like *--nodocs*.
```

Comment:
Hmm, this is a bit confusing - it can be read as "This will accurately reflect the `--nodocs` options", meaning that it will *not* iterate over those skipped files. Maybe use "print" instead of "reflect"?

Reply:
Eh, that wouldn't be good either:

This will accurately print any files skipped due to options like *--nodocs*.

Dunno, maybe no need to get hung up on this :smile:

Reply:
Yeah, the point is that non-installed files are NOT in the output. So I kinda prefer the reflected term here - the output reflects file install policies.

Reply:
... but indeed, the reworded version is even better :+1:

Reply:
Minor correction: You would still need the queryformat variant so that you could actually use the `%{instfilenames}` macro, since it's a runtime script, not a build script, of course.